src/daft-plan/src/builder.rs

```diff
@@ -26,15 +26,18 @@ impl LogicalPlanBuilder {
 #[pymethods]
 impl LogicalPlanBuilder {
     #[staticmethod]
-    pub fn source(filepaths: Vec<String>, schema: &PySchema) -> PyResult<LogicalPlanBuilder> {
-        let source_info = source_info::SourceInfo::FilesInfo(source_info::FilesInfo::new(
-            filepaths,
-            schema.schema.clone(),
-        ));
+    pub fn read_parquet(filepaths: Vec<String>, schema: &PySchema) -> PyResult<LogicalPlanBuilder> {
```

Comment:
~Nit: Instead of having a `LogicalPlanBuilder` method per read API (e.g. for reading Parquets, CSVs, JSONs, etc.), we could still have a `source` method that also takes a `FileFormat` enum variant, which would then be incorporated into the `SourceInfo`.~

On second thought, making the `LogicalPlanBuilder` methods 1:1 with the `DataFrame` APIs makes more sense to me than 1:1 with the logical operators, since the former leaks less representational details to the Python side and makes the builder abstraction more useful so keeping format-specific `read_*` methods seems like the best choice to me! E.g. it would also allow us to hide a `FileFormat` enum from the Python side.